### PR TITLE
Ransoming as a Pirate is more favorable

### PR DESCRIPTION
**Pull request by SgtHunk**
## About The Pull Request

Now there are a few more variables in pirate ransoming, and some notable changes.
The main one being: you can ransom any human player, **dead or alive**.
Before, this was simply any alive human player but it's VERY inconvenient to ransom people when your shuttle is in space and you have to deal with:
- Turrets targeting your ransom target (which could end up hitting you)
- The atmospheric conditions until you get to the pad (space and obviously the shuttle won't have good atmos conditions 9/10 times)
- Having to capture your target (ideally without any harm???) as a pirate when all you have are extremely lethal options like eswords, laser guns and shotguns.

This results in ransoming being rarely seen simply because most of the time your target will be dead by the time you get to the pad.

As for the variable changes:
The base value for a ransom is 500.
If the target is alive, this value is multiplied by 2.
If the target is security, this value is multiplied by 2.
If the target is command, this value is multiplied by 5.

For some examples: An alive captain will give an export value of 5000, a dead warden will give an export value of 1000, an alive security officer a value of 2000, and so on.
Let me know if you think the values are too extreme, but I found it fair when you consider that the amount of heads are limited and I would be surprised if a pirate manages to export all of them for that sweet cash. Security gets a bonus because they're harder targets to capture.

## Why It's Good For The Game

The pirate ransom export is very rarely seen, and making it so it allows dead humans will greatly increase the chances of seeing this, it encourages pirates to be more aggressive against certain targets (security and command) with the intention of getting them ransomed for a good sum.
The current loop of pirates of raiding ore silo or a lathe gets a bit stale over time, and they should be able to go for a more aggressive and direct route that targets crew that is less awkward than mugging someone for their bank account holochips.

## Changelog

:cl:
balance: Pirates can now ransom any crew member, dead or alive, for cash towards their objective. Ransom special targets like security and command for a bigger reward. Simply place the crew member on the cargo pad, and send away!
/:cl:

### CODE REVIEW

**Comment by SkeletalElite** [comment]
Will HoS be more valuable than captain since he's both command and security? If so maybe hte captain should get his own special multipler as well, like 3x for being the captain. That way an alive HoS is 10,000 and an alive captain is 15,000

**Comment by Ryusenshu** [comment]
Dont Pirates still qdel every person teled with the bounty pad?

**Comment by Profakos** [comment]
Yeah, ransommed people flat out get deleted. Would be cool if they were sent to an actual cargo hold z level, syndicate kidnapping base, with piles of decorative gold piles they can't touch, but that is probably subject for a different PR.

**Comment by carlarctg** [comment]
It should use the syndicate ransom base and send them back down with a pax skillchip

**Comment by HoodMilk** [comment]
encouraging pirates to delete people is lame

File: code/modules/antagonists/pirate/pirate_shuttle_equipment.dm [hunk 1]
```diff
@@ -352,23 +352,25 @@
...
+	if(!hostage.mind || (FACTION_PIRATE in hostage.faction))
 		return 0
-	else if(FACTION_PIRATE in H.faction) //can't ransom your fellow pirates to CentCom!
-		return 0
-	else if(H.mind.assigned_role.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND)
-		return 3000
-	else
-		return 1000
+	var/ransom_value = 500
+	if(hostage.stat == CONSCIOUS)
```
Comment (Fikou) [inline code/modules/antagonists/pirate/pirate_shuttle_equipment.dm:367]: what if theyre sleeping or somethin

Comment (BlueMemesauce) [reply code/modules/antagonists/pirate/pirate_shuttle_equipment.dm:367]: and crit people are alive too
unless they have to be in "pristine condition" like before

**Comment by itseasytosee** [comment]
pirates not deleting people is lame. although storing them in nullspace with a way to retrieve them is cool.

**Comment by BlueMemesauce** [comment]
> pirates not deleting people is lame. although storing them in nullspace with a way to retrieve them is cool.

just copy the mechanic for the traitor kidnapping objective where you get sent to Japan for a short amount of time

**Comment by SpaceVampire** [comment]
I'm just going to agree with everyone else here.
Pirates being able to actually use the ransoming feature is good.
But having it just delete people from the round is bad.
Kidnapping and heretics send people to special zones instead for a reason.

**Comment by DrDiasyl** [comment]
We should replace pirates deleting people with something else, before making it more favourable for pirates.

**Comment by itseasytosee** [comment]
temporary gay baby jails fucking suck. delete them or make a way to rescue them.
heretic shadow realm and ninja prison also both suck

File: code/modules/antagonists/pirate/pirate_shuttle_equipment.dm (hunk 1 above)
Comment (SgtHunk) [reply code/modules/antagonists/pirate/pirate_shuttle_equipment.dm]: changed on latest commit

**Comment by SgtHunk** [comment]
> Will HoS be more valuable than captain since he's both command and security?
HOS is in fact worth 15k points (alive) due to the way the conditions apply, which means fighting security has a pretty good payout for a considerable risk.

> Dont Pirates still qdel every person teled with the bounty pad?
They do, things like ninja prison have existed long before pirate exports, but it makes sense because things sent through cargo, when exported, do just get deleted upon arrival, and I assume it was easier to just keep it that way than add a lot of code to account for one special type of export.

**Comment by SgtHunk** [comment]
Contractor simply removed a random amount of creds from the cargo budget and sent them back after a set amount of time, ninja literally never sent them back so if you got captured you were stuck all round on prison

File: code/modules/antagonists/pirate/pirate_shuttle_equipment.dm [hunk 2]
```diff
@@ -364,23 +364,25 @@
...
+	if(!ransomee.mind || (FACTION_PIRATE in ransomee.faction))
 		return 0
-	else if(FACTION_PIRATE in ransomee.faction) //can't ransom your fellow pirates to CentCom!
-		return 0
-	else if(ransomee.mind.assigned_role.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND)
-		return 3000
-	else
-		return 1000
+	var/ransom_value = 500
+	if(!ransomee.stat == DEAD)
```
Comment (Fikou) [inline code/modules/antagonists/pirate/pirate_shuttle_equipment.dm:379]: !ransomee.stat will never be equal to 4, it will either be 0 or 1

**Comment by CMDR-Gungnir** [comment]
Definitely change it so Pirates don't just delete people when they do the ransom, if we're going to encourage it more.

Doesn't make any sense, anyway. They're ransoming you, but they also never give you back??? Doesn't sound like a ransom at all, to me.

File: code/modules/antagonists/pirate/pirate_shuttle_equipment.dm [hunk 3]
```diff
@@ -364,23 +364,25 @@
...
-	else if(FACTION_PIRATE in ransomee.faction) //can't ransom your fellow pirates to CentCom!
-		return 0
-	else if(ransomee.mind.assigned_role.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND)
-		return 3000
-	else
-		return 1000
+	var/ransom_value = 500
+	if(ransomee.stat != DEAD)
+		ransom_value *= 2
+	if(ransomee.mind.assigned_role.departments_bitflags & DEPARTMENT_BITFLAG_SECURITY)
```
Comment (ShizCalev) [inline code/modules/antagonists/pirate/pirate_shuttle_equipment.dm:381]: null check assigned_role

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
not a fan of this with the current iteration that it just lol deletes you (otherwise id be down)

**Comment by YehnBeep** [comment]
> The pirate ransom export is very rarely seen, and making it so it allows dead humans will greatly increase the chances of seeing this, it encourages pirates to be more aggressive against certain targets (security and command)

I feel like pirates should not be mae to be more of an aggressive "RRs you" antagonist, personally. It's not like there aren't enough of those already.

> and they should be able to go for a more aggressive and direct route that targets crew that is less awkward than mugging someone for their bank account holochips.

...But I feel like that's sort of the very definition of pirates. They may kill you but that's incidental to them trying to gain wealth. Incentivizing them to essentially death match with an extra step (and said step deleting people)... I'm not a fan.

**Comment by ExcessiveUseOfCobblestone** [comment]
Re-PR once ransoming doesn't just lol delete you and I'm all for it

**Pull request closed without merging**